recipes/categorical_anomaly.py

```diff
@@ -228,11 +333,16 @@ def graph_highd(self, three_d: bool = None, **kwargs):
                     pca = PCA(n_components=2)
                     components = pca.fit_transform(feat_vals)
                     loadings = pca.components_.T * np.sqrt(pca.explained_variance_)
-                    score_df = pd.concat([self.score_df, pd.DataFrame(components)], axis=1)
+                    score_df = pd.concat([self.score_df.reset_index(), pd.DataFrame(components)], axis=1).reset_index()
                     score_df = score_df.rename(columns={0:'PC1', 1:'PC2'})
 
-                    fig = px.scatter(score_df, x='PC1', y='PC2', title='PCA Plot - Anomalies',
+                    # Append values to non-scaled df so users can see
+                    # the normal values for their features graphed
+                    scores = self.df.reset_index().join(score_df[['score', 'anomaly', 'PC1', 'PC2',]])
```

Comment:
Looks like line `341~362` are  same as line `309~329` (except line `315` and `348`), could we move part of the else condition clause (line 332-336) above the line 309 to make the code more concise

Reply:
Ya I think I was having trouble getting a simple way around that numeric_functions to metrics_column thing, it's such an edge case, but once I started with the logic to include it, I needed it in every function.
Just changed it to make it more concise by doing an if statement, then changing self.metrics_column to the numeric features if that logic is triggered.